src/main.rs

```diff
@@ -496,7 +496,33 @@ fn add_to_autostart() {
 		.join(".config/autostart/TimePlot.desktop");
 	ensure_file(&file_path, &xdg_desktop);
 }
-#[cfg(not(target_os = "linux"))]
+
+#[cfg(target_os = "macos")]
+fn add_to_autostart() {
+	let executable_name = env::current_exe().expect("failed to get current executable name");
+	let executable_name = executable_name.to_str().unwrap_or_else(|| {
+		panic!(
+			"failed to read executable name '{:?}' to string",
+			executable_name
+		)
+	});
```

Comment:
If the code is identical between the "macos" and other OS-es, maybe move it to a separate function, e.g.
```
fn executable_name()
```
?